code/modules/antagonists/heretic/magic/nightwatcher_rebirth.dm

```diff
@@ -13,11 +13,12 @@
 	action_icon = 'icons/mob/actions/actions_ecult.dmi'
 	action_icon_state = "smoke"
 
-/obj/effect/proc_holder/spell/targeted/fiery_rebirth/cast(list/targets, mob/user)
-	if(!ishuman(user))
+/obj/effect/proc_holder/spell/targeted/fiery_rebirth/cast(list/targets, mob/living/carbon/human/user)
+	if(!istype(user))
+		revert_cast()
 		return
-	var/mob/living/carbon/human/human_user = user
-	human_user.ExtinguishMob()
+	var/did_something = user.on_fire
+	user.ExtinguishMob()
```

Comment:
This can Extinguish items even if `user.on_fire` is a positive value, which means it DOES something.
I won't ask you to change the whole proc logic, but you need to put a comment.

Reply:
True, although I believe holding flaming items while not being on fire yourself is quite rare

Reply:
yeah just put a comment. That works enough. It's easy to miss when someone wants to change.